content/docs/rosa/alb-tgw-nlb-e2e/index.md

```diff
@@ -6,95 +7,167 @@ authors:
 
-### Check TLS at the NLB layer
+1. Run the following command to get your subnets
 
-```bash
-export NLB_URL=$(oc get svc -n ech-server http-https-echo -ojsonpath='{.status.loadBalancer.ingress[0].hostname}')
-echo "NLB's FQDN:  $NLB_URL"
-export NLB_IP=$(dig +short $NLB_URL | head -1)
-echo "NLP IP address $NLB_IP"
-```
-Check application  endpoint
+    ```bash
+    aws ec2 describe-subnets --region ${REGION} \
+      --subnet-ids $(echo "${SUBNETS}") \
+      --query 'Subnets[*].CidrBlock' --output text
+    ```
 
-```bash
- curl -v -k --resolve secured-echo-server.com:443:$NLB_IP https://secured-echo-server.com:443/productpage
-```
+    ```
+    10.0.16.0/22    10.0.12.0/22    10.0.20.0/22
+    ```
+
+1. Pick a an unused IP address from each of these CIDRs (you may need to use the AWS console to do this, although its a good bet a high IP in each CIDR is free).
 
-### Create an ALB in the public subnet 
+    > In this example we'll use `10.0.16.200`, `10.0.12.200`, and `10.0.20.200`.
 
-We need to create an ALB in the ingress/egress VPC. To do this, we first need to create a Target Group (TG) within this VPC and then associate this TG with the ALB's targets. To create the TG, we require the IP addresses of the NLB. 
+1. Edit the file `./apps/echo-server/echo-server.yaml` and find the line that contains the text `service.beta.kubernetes.io/aws-load-balancer-private-ipv4-addresses`.  Replace the example IPs with the ones you picked above.
+
+1. Create a project to run the echo server application in
+
+    ```bash
+    oc new-project echo-server
+    ```
+
+1. Deploy the echo server application
+
+    ```bash
+    oc apply -f ./apps/echo-server/echo-server.yaml
+    ```
+
+1. Wait until the deployment is running
+
+    ```bash
+    oc rollout status deployment/http-https-echo
+    ```
+
+1. Wait until the service has a Load Balancer attached
+
+    > Hint: the `EXTERNAL-IP` field should contain a DNS record, if not, wait a minute and run this command again.
+
+    ```bash
+    oc get service
+    ```
+
+    ```
+    NAME              TYPE           CLUSTER-IP     EXTERNAL-IP
+        PORT(S)                         AGE
+    http-https-echo   LoadBalancer   172.30.52.80   k8s-echoserv-httphttp-12c646e3a6-58ae04e547de7af0.elb.us-east-1.amazonaws.com   8080:30127/TCP,8443:32553/TCP   24s
+    ```
+
+1. Get the URL of the NLB
+
+    ```bash
+    export NLB_URL=$(oc get svc -n echo-server http-https-echo -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+    echo "NLB's FQDN: $NLB_URL"
+    ```
+
+1. Check you can access the endpoint
+
+    > Note: The NLB inherits the security group of its targets, which in this case means you need to access it from inside the subnets that the ROSA worker nodes are in.  We do this by running the command through `oc debug`
+
+    ```bash
+    NODE=$(oc get nodes -l 'node-role.kubernetes.io/worker=' \
+      -o name | head -1)
+    oc debug $NODE -- curl -s -v -k https://$NLB_URL
```

Comment:
run the curl inside a worker node,  note we don't need any of the resolve stuff, again its just an NLB to a service.